documentation/library_guide/parallel_api/pass_data_algorithms.rst

```diff
@@ -56,72 +62,91 @@ To use the functions, add ``#include <oneapi/dpl/iterator>`` to your code. For e
 Use Unified Shared Memory
 -------------------------
 
-The following examples demonstrate two ways to use the parallel algorithms with USM:
-
-* USM pointers
-* USM allocators
-
-If you have a USM-allocated buffer, pass the pointers to the start and past the end
-of the buffer to a parallel algorithm. Make sure that the execution policy and
-the buffer were created for the same queue. For example:
+If you have USM-allocated memory, pass the pointers to the start and past the end
+of the sequence to a parallel algorithm. Make sure that the execution policy and
+the USM-allocated memory were created for the same queue. For example:
 
 .. code:: cpp
 
   #include <oneapi/dpl/execution>
   #include <oneapi/dpl/algorithm>
+  #include <random>
   #include <sycl/sycl.hpp>
   int main(){
     sycl::queue q;
     const int n = 1000;
     int* d_head = sycl::malloc_shared<int>(n, q);
+    std::generate(d_head, d_head + n, std::minstd_rand{});
 
-    std::fill(oneapi::dpl::execution::make_device_policy(q), d_head, d_head + n, 42);
+    std::sort(oneapi::dpl::execution::make_device_policy(q), d_head, d_head + n);
 
     sycl::free(d_head, q);
     return 0;
   }
 
-Alternatively, use ``std::vector`` with a USM allocator. For example:
+When using device USM, such as allocated by malloc_device, the user is responsible for data
+transfers to and from the device to ensure that input data is device accessible during oneDPL
```

Comment:
My suggestion:

When using device USM, such as allocated by ``malloc_device``, you are responsible for data
transfers to and from the device to ensure that the input data is device accessible during oneDPL
algorithm execution and that the result is available to the operation.

Reply:
Taken, but pluralizing "operations" at the end.

Reply:
Looks good! Should malloc_device be in monospace font? It is not in the original text.

Reply:
good catch, fixed.